### PR TITLE
feat: add deno as package manager

### PR DESCRIPTION
**Pull request by bartlomieju**
### Description

Add Deno as a valid package manager recognized by `package-manager-detector`.

### Linked Issues

Closes https://github.com/antfu-collective/package-manager-detector/issues/21

### Additional context

As of Deno v2.0.5, Deno will provide `npm_config_user_agent` env var in relevant subcommands (https://github.com/denoland/deno/pull/26639).

### CODE REVIEW

File: src/commands.ts [hunk 1]
```diff
@@ -54,6 +54,21 @@ const bun: AgentCommands = {
...
 
+const deno: AgentCommands = {
+  'agent': ['deno', 0],
+  'run': ['deno', 'run', 0],
+  'install': ['deno', 'install', 0],
+  'frozen': ['deno', 'install', '--frozen'],
+  'global': ['deno', 'install', '-g', 0],
+  'add': ['deno', 'add', 0],
+  'upgrade': ['deno', 'update', 0],
+  'upgrade-interactive': ['deno', 'update', 0],
```
Comment (bartlomieju) [inline src/commands.ts:65]: This is actually not true yet, we plan to add these commands in Deno v2.1 later in November.

File: test/__snapshots__/detect-sync.spec.ts.snap [hunk 2]
```diff
@@ -52,6 +59,14 @@ exports[`packager > bun 1`] = `
 }
 `;
 
+exports[`packager > deno 1`] = `
+{
+  "agent": "bun",
+  "name": "bun",
+  "version": "0",
```
Comment (bartlomieju) [inline test/__snapshots__/detect-sync.spec.ts.snap:66]: This doesn't look correct

File: test/__snapshots__/detect.spec.ts.snap [hunk 3]
```diff
@@ -52,6 +59,14 @@ exports[`packager > bun 1`] = `
 }
 `;
 
+exports[`packager > deno 1`] = `
+{
+  "agent": "bun",
+  "name": "bun",
+  "version": "0",
```
Comment (bartlomieju) [inline test/__snapshots__/detect.spec.ts.snap:66]: Ditto

**Review by benmccann — APPROVED** [review]

File: src/commands.ts [hunk 4]
```diff
@@ -54,6 +54,21 @@ const bun: AgentCommands = {
...
+  'execute': ['deno', 'run', 0],
+  'execute-local': ['deno', 'run', 0],
```
Comment (benmccann) [inline src/commands.ts:67]: It looks like the same command is used here for `execute` and `execute-local`, which means this isn't working in in the Svelte CLI: https://github.com/sveltejs/cli/pull/313#pullrequestreview-2443372777. Does Deno have some equivalent to `pnpm exec` or should we make a feature request for that?

Comment (bartlomieju) [reply src/commands.ts:67]: Ooops, my bad. We currently have no way to do it, so a feature request would be helpful. I think we could handle that with `deno task --eval` so it uses cross-platform shell.

Comment (benmccann) [reply src/commands.ts:67]: Okay, filed a request here: https://github.com/denoland/deno/issues/26918

**Pull request merged**